src/extension.ts

```diff
@@ -847,6 +849,28 @@ function createMonitor(): any {
     });
 }
 
+function createIdfTerminal() {
+    PreCheck.perform([webIdeCheck, openFolderCheck], () => {
+        const envSetCmd = process.platform === "win32" ? "set" : "export";
+        const modifiedEnv = utils.appendIdfAndToolsToPath();
+        const espIdfTerminal = vscode.window.createTerminal({ name: "ESP-IDF Terminal", env: modifiedEnv,
+                                                              cwd: workspaceRoot.fsPath });
+        espIdfTerminal.show();
+        espIdfTerminal.sendText(`${envSetCmd} IDF_PATH=${modifiedEnv.IDF_PATH}`);
+        let IDF_ADD_PATHS_EXTRAS =
+            `${path.join(modifiedEnv.IDF_PATH, "components", "esptool_py", "esptool")}`;
+        IDF_ADD_PATHS_EXTRAS =
+            `${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${path.join(modifiedEnv.IDF_PATH,
+                                                                 "components", "espcoredump")}`;
+        IDF_ADD_PATHS_EXTRAS =
+            `${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${path.join(modifiedEnv.IDF_PATH,
+                                                                 "components", "partition_table")}`;
+        espIdfTerminal.sendText(`${envSetCmd} PATH=${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${modifiedEnv.PATH}`);
```

Comment:
PATH environment variable (at least on Windows) is very likely to contain spaces. You need to add quotes, differently for bash and for Windows CMD:
```
export PATH="${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${modifiedEnv.PATH}"
set "PATH=${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${modifiedEnv.PATH}"
```
```